### PR TITLE
Implement collection fe

### PR DESCRIPTION
**Pull request by muhajirdev**

### CODE REVIEW

**Comment by mmahalwy** [comment]
#### 📝 Changed routes:


| [**…ectionId\]/index.module**](https://quran-d5clm6srn-qurancom.vercel.app/collections/[collectionId]/index.module) | [**…ections/\[collectionId\]**](https://quran-d5clm6srn-qurancom.vercel.app/collections/[collectionId]) |
|:-:|:-:|
|<a href="https://quran-d5clm6srn-qurancom.vercel.app/collections/[collectionId]/index.module">Screenshot of /collections/[collectionId]/index.module</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/dw5sILhcBOJRtjBOZFvHT_7R0jKUwoYSUlEkH2PBMIGHAmO4Wg8_5fljZQimetfL4OoSFRfFK3BqF58k3aEMGg.png">(view full size)</a> |<a href="https://quran-d5clm6srn-qurancom.vercel.app/collections/[collectionId]">Screenshot of /collections/[collectionId]</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/25OwY3am0rQZiIgxSKVT4Dw1GNu9m3Rnj9kope3NSCW8sS4OtjTqOMdibfjskQGoa4Ggs52QPPNpfo_ta1VBBA.png">(view full size)</a> |



| [**/profile.module**](https://quran-d5clm6srn-qurancom.vercel.app/profile.module) | [**/profile**](https://quran-d5clm6srn-qurancom.vercel.app/profile) |
|:-:|:-:|
|<a href="https://quran-d5clm6srn-qurancom.vercel.app/profile.module">Screenshot of /profile.module</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/4RuiHAhVZzSbCMDoD6Qu3qhdi85y1Lmmlu-1cvdLRg7GskAIcs80ZS6Xj1Zw5LrsCeQgXVWc9Z5q_H_gSBnAqA.png">(view full size)</a> |<a href="https://quran-d5clm6srn-qurancom.vercel.app/profile">Screenshot of /profile</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/DHB7hSDeC7L_J2esGmvjMKSsLHTwSkK0MJqVX_hEhT4tJbl3uvCEL5QlETT50GUANLYWtPIbSeXeGk70NmhDXA.png">(view full size)</a> |







Commit 13a829d50027b8f8fe6ecac1fd78a85dfa3dc4ea (https://quran-d5clm6srn-qurancom.vercel.app).

File: src/components/Collection/CollectionDetail/CollectionDetail.tsx [hunk 1]
```diff
@@ -1,32 +1,76 @@
...
+
+  const onBookmarkItemDeleted = (bookmarkId: string) => {
+    deleteCollectionBookmark(id, bookmarkId).then(() => {
+      onUpdated();
+    });
+  };
+
+  const sortOptions = [
+    {
+      id: 'recentlyAdded',
```
Comment (muhajirdev) [inline src/components/Collection/CollectionDetail/CollectionDetail.tsx:62]: id can be an enum

File: src/components/Collection/CollectionList/CollectionList.tsx [hunk 2]
```diff
@@ -1,77 +1,108 @@
...
+  const apiParams = {
+    sortBy,
+  };
+
+  const { data, mutate } = useSWR<any>(makeCollectionsUrl(apiParams), () =>
+    getCollectionsList(apiParams),
   );
+
+  const sortOptions = [
+    { id: 'recentlyUpdated', label: t('collection:recently-updated') },
```
Comment (muhajirdev) [inline src/components/Collection/CollectionList/CollectionList.tsx:34]: id can be an enum

File: src/components/QuranReader/TranslationView/QuranReflectButton.tsx [hunk 3]
```diff
@@ -34,12 +34,12 @@ const QuranReflectButton = ({
...
 
-  // const onButtonClicked = () => {
-  //   // eslint-disable-next-line i18next/no-literal-string
-  //   logButtonClick(`${isTranslationView ? 'translation_view' : 'reading_view'}_reflect`);
-  //   setIsContentModalOpen(true);
-  //   fakeNavigate(getVerseReflectionNavigationUrl(verseKey), lang);
-  // };
+  const onButtonClicked = () => {
+    // eslint-disable-next-line i18next/no-literal-string
+    // logButtonClick(`${isTranslationView ? 'translation_view' : 'reading_view'}_reflect`);
```
Comment (muhajirdev) [inline src/components/QuranReader/TranslationView/QuranReflectButton.tsx:39]: revert this file to the one on master

File: src/components/Verse/SaveToCollectionAction.tsx [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+import { logButtonClick } from 'src/utils/eventLogger';
+import BookmarkType from 'types/BookmarkType';
+
+const SaveToCollectionAction = ({ verse, bookmarksRangeUrl }) => {
+  const [isSaveCollectionModalOpen, setIsSaveCollectionModalOpen] = useState(false);
+  const quranReaderStyles = useSelector(selectQuranReaderStyles, shallowEqual);
+  const mushafId = getMushafId(quranReaderStyles.quranFont, quranReaderStyles.mushafLines).mushaf;
+  const { t } = useTranslation();
+  const { data: collectionListData, mutate: mutateCollectionListData } = useSWR(
+    makeCollectionsUrl({ limit: 50 }),
```
Comment (muhajirdev) [inline src/components/Verse/SaveToCollectionAction.tsx:40]: limit 50 should be removed

**Comment by mmahalwy** [comment]
#### 📝 Changed routes:


| [**…ectionId\]/index.module**](https://quran-dv1n4rlvc-qurancom.vercel.app/collections/[collectionId]/index.module) | [**…ections/\[collectionId\]**](https://quran-dv1n4rlvc-qurancom.vercel.app/collections/[collectionId]) |
|:-:|:-:|
|<a href="https://quran-dv1n4rlvc-qurancom.vercel.app/collections/[collectionId]/index.module">Screenshot of /collections/[collectionId]/index.module</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/9t2inyWfG-N8YqCLg3SwYYBiv0w7mWFOQtxk7tc2G3xuiu9HjbTQxXamfRLsZopTEuT77O8D3fgF_Q0VnUYJJw.png">(view full size)</a> |<a href="https://quran-dv1n4rlvc-qurancom.vercel.app/collections/[collectionId]">Screenshot of /collections/[collectionId]</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/pAL_FPfmMRcRp8RE4CyJjM-kt1qzTJ1A2G1LluHTGQAjUWj2-EATGecGKr08c_vICfzmSzw3rA-EgzTtoQAb4Q.png">(view full size)</a> |



| [**/profile.module**](https://quran-dv1n4rlvc-qurancom.vercel.app/profile.module) | [**/profile**](https://quran-dv1n4rlvc-qurancom.vercel.app/profile) |
|:-:|:-:|
|<a href="https://quran-dv1n4rlvc-qurancom.vercel.app/profile.module">Screenshot of /profile.module</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/4iMc6X694Pqh61pxK4zK28OKbw4SpmGVabZCfvsTFPHZU1Oa3tG4Jc-MQy3fVerEFcAlAXaX21lq5A3natA2KQ.png">(view full size)</a> |<a href="https://quran-dv1n4rlvc-qurancom.vercel.app/profile">Screenshot of /profile</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/lnnShDg-IWryogTXLTPnv4z4W5gS5Sy_-BAWZj3vSDsksBWrcG_xQ1EliCZmpFn9GzAYVsDZ2uxnb0dQYQ0MxQ.png">(view full size)</a> |







Commit 9660daa96ab59eab8f2478d6f0905aaaf0179f61 (https://quran-dv1n4rlvc-qurancom.vercel.app).

File: src/components/QuranReader/TafsirView/TafsirVerseText/index.tsx [hunk 5]
```diff
@@ -9,7 +9,7 @@ import { getVerseWords } from 'src/utils/verse';
 import Verse from 'types/Verse';
 
 interface Props {
-  verses: Record<string, Verse>;
+  verses: Verse[];
 }
 
 const TafsirVerseText: React.FC<Props> = ({ verses }) => {
```
Comment (osamasayed) [inline src/components/QuranReader/TafsirView/TafsirVerseText/index.tsx:15]: Since this is becoming a general-purpose component and not related to Tafisr anymore, we should rename it to maybe something like `VerseTextPreview` or something similar and move it up in the directory tree.

File: src/components/Collection/CollectionList/DeleteCollectionAction.tsx [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+const DeleteCollectionAction = ({ collectionId, onDone }) => {
+  const { t } = useTranslation();
+
+  const onMenuClicked = () => {
+    deleteCollection(collectionId).then(() => {
+      onDone();
+    });
+  };
+
+  return <PopoverMenu.Item onClick={onMenuClicked}>{t('profile:delete')}</PopoverMenu.Item>;
```
Comment (osamasayed) [inline src/components/Collection/CollectionList/DeleteCollectionAction.tsx:15]: There is an extra ";"

File: src/components/Collection/CollectionList/DeleteCollectionAction.tsx [hunk 7]
```diff
@@ -0,0 +1,18 @@
+import useTranslation from 'next-translate/useTranslation';
+
+import PopoverMenu from 'src/components/dls/PopoverMenu/PopoverMenu';
+import { deleteCollection } from 'src/utils/auth/api';
+
+const DeleteCollectionAction = ({ collectionId, onDone }) => {
+  const { t } = useTranslation();
+
+  const onMenuClicked = () => {
```
Comment (osamasayed) [inline src/components/Collection/CollectionList/DeleteCollectionAction.tsx:9]: nit: `omMenuItemClicked`.

File: src/components/Collection/CollectionList/RenameCollectionAction.tsx [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+import RenameCollectionModal from '../RenameCollectionModal/RenameCollectionModal';
+
+import PopoverMenu from 'src/components/dls/PopoverMenu/PopoverMenu';
+import { updateCollection } from 'src/utils/auth/api';
+
+const RenameCollectionAction = ({ currentCollectionName, collectionId, onDone }) => {
+  const { t } = useTranslation();
+  const [isModalOpen, setIsModalOpen] = useState(false);
+
+  const onMenuClicked = () => {
```
Comment (osamasayed) [inline src/components/Collection/CollectionList/RenameCollectionAction.tsx:14]: Ditto: `onMenuItemClicked`.

File: src/components/Collection/CollectionList/DeleteCollectionAction.tsx (hunk 7 above)
Comment (osamasayed) [inline src/components/Collection/CollectionList/DeleteCollectionAction.tsx:9]: nit: `onMenuItemClicked`.

File: src/components/Collection/CollectionList/RenameCollectionAction.tsx [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+    setIsModalOpen(true);
+  };
+
+  const onSubmit = (data) => {
+    updateCollection(collectionId, { name: data.name }).then(() => {
+      onDone();
+      setIsModalOpen(false);
+    });
```
Comment (osamasayed) [inline src/components/Collection/CollectionList/RenameCollectionAction.tsx:22]: What happens when there is an issue with the API? I think we should show a toast/error message

File: src/components/Collection/CollectionList/DeleteCollectionAction.tsx [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+    deleteCollection(collectionId).then(() => {
+      onDone();
+    });
```
Comment (osamasayed) [inline src/components/Collection/CollectionList/DeleteCollectionAction.tsx:12]: What happens when there is an issue with the API? I think we should show a toast/error message

File: src/components/Collection/SaveToCollectionModal/SaveToCollectionModal.tsx [hunk 11]
```diff
@@ -75,7 +77,7 @@ const SaveToCollectionModal = ({
                 ]}
                 actionText={t('common:submit')}
                 onSubmit={(data: any) => {
-                  onNewCollectionCreated(data.name);
+                  onNewCollectionCreated(data.name).then(() => setIsAddingNewCollection(false));
```
Comment (osamasayed) [inline src/components/Collection/SaveToCollectionModal/SaveToCollectionModal.tsx:80]: What happens when there is an issue with the API? I think we should show a toast/error message

File: src/components/Collection/CollectionDetail/CollectionDetail.tsx [hunk 12]
```diff
@@ -1,51 +1,178 @@
...
+  const { t, lang } = useTranslation();
+  const quranReaderStyles = useSelector(selectQuranReaderStyles, shallowEqual);
+  const { quranFont, mushafLines } = quranReaderStyles;
+  const { mushaf } = getMushafId(quranFont, mushafLines);
+  const selectedTranslations = useSelector(selectSelectedTranslations, areArraysEqual);
+
+  const onBookmarkItemDeleted = (bookmarkId: string) => {
+    deleteCollectionBookmark(id, bookmarkId).then(() => {
+      onUpdated();
+    });
```
Comment (osamasayed) [inline src/components/Collection/CollectionDetail/CollectionDetail.tsx:66]: What happens when there is an issue with the API? I think we should show a toast/error message

File: src/pages/collections/[collectionId]/index.tsx [hunk 13]
```diff
@@ -0,0 +1,122 @@
...
+    }
+
+    const cursor = previousPageData.pagination?.endCursor;
+    return makeGetBookmarkByCollectionId(collectionId, {
+      sortBy,
+      cursor,
+    });
+  };
+
+  const { data, size, setSize, mutate } = useSWRInfinite<GetBookmarkCollectionsIdResponse>(
```
Comment (osamasayed) [inline src/pages/collections/[collectionId]/index.tsx:57]: We need to add a loading indicator because currently while the data is loading, the body becomes empty and once the response is there, there is a major layout shift.

File: src/components/Collection/SaveToCollectionModal/SaveToCollectionModal.tsx [hunk 14]
```diff
@@ -75,7 +77,7 @@ const SaveToCollectionModal = ({
                 ]}
                 actionText={t('common:submit')}
                 onSubmit={(data: any) => {
```
Comment (osamasayed) [inline src/components/Collection/SaveToCollectionModal/SaveToCollectionModal.tsx:79]: nit: we should not make this an in-line function

File: src/components/Collection/CollectionDetail/CollectionDetail.tsx [hunk 15]
```diff
@@ -1,51 +1,178 @@
...
-  collectionItems: CollectionItem[];
+  bookmarks: Bookmark[];
+  sortBy: string;
+  onSortByChange: (sortBy: string) => void;
+  onUpdated: () => void;
 };
 
-const CollectionDetail = ({ title, collectionItems }: CollectionDetailProps) => {
-  const { t } = useTranslation();
+enum CollectionDetailSortOption {
```
Comment (osamasayed) [inline src/components/Collection/CollectionDetail/CollectionDetail.tsx:44]: nit: this should be extracted out

File: src/components/Collection/SaveToCollectionModal/SaveToCollectionModal.tsx [hunk 16]
```diff
@@ -13,40 +13,42 @@ import { RuleType } from 'types/FieldRule';
...
         <div className={styles.collectionList}>
           {collections.map((collection) => (
             <div className={styles.collectionItem} key={collection.id}>
               <Checkbox
                 id={collection.name}
-                checked={collection.checked}
+                defaultChecked={collection.checked}
                 label={collection.name}
-                onChange={() => onCollectionToggled(collection)}
+                onChange={(checked) => onCollectionToggled(collection, checked)}
```
Comment (osamasayed) [inline src/components/Collection/SaveToCollectionModal/SaveToCollectionModal.tsx:51]: nit: we shouldn't make it an in-line function

File: src/components/Collection/CollectionList/CollectionList.tsx [hunk 17]
```diff
@@ -1,76 +1,113 @@
...
-  id: string | number;
-  name: string;
-  itemCount: number;
-};
+const defaultSortOptionId = 'recentlyUpdated';
 
-type CollectionListProps = {
-  collections: Collection[];
-};
+enum CollectionListSortOption {
```
Comment (osamasayed) [inline src/components/Collection/CollectionList/CollectionList.tsx:22]: nit: this should be extracted out into a file.

File: src/components/Collection/CollectionList/CollectionList.tsx [hunk 18]
```diff
@@ -1,76 +1,113 @@
...
+import PopoverMenu from 'src/components/dls/PopoverMenu/PopoverMenu';
+import { getCollectionsList } from 'src/utils/auth/api';
+import { makeCollectionsUrl } from 'src/utils/auth/apiPaths';
 
-type Collection = {
-  id: string | number;
-  name: string;
-  itemCount: number;
-};
+const defaultSortOptionId = 'recentlyUpdated';
```
Comment (osamasayed) [inline src/components/Collection/CollectionList/CollectionList.tsx:20]: This should be using the enum instead of a string value. Also since it's fixed we should set it as `DEFAULT_SORT_OPTION`.

File: src/components/Collection/CollectionDetail/CollectionDetail.tsx [hunk 19]
```diff
@@ -1,51 +1,178 @@
...
+  onUpdated,
+}: CollectionDetailProps) => {
+  const { t, lang } = useTranslation();
+  const quranReaderStyles = useSelector(selectQuranReaderStyles, shallowEqual);
+  const { quranFont, mushafLines } = quranReaderStyles;
+  const { mushaf } = getMushafId(quranFont, mushafLines);
+  const selectedTranslations = useSelector(selectSelectedTranslations, areArraysEqual);
+
+  const onBookmarkItemDeleted = (bookmarkId: string) => {
+    deleteCollectionBookmark(id, bookmarkId).then(() => {
```
Comment (osamasayed) [inline src/components/Collection/CollectionDetail/CollectionDetail.tsx:64]: Add logging

File: src/components/Collection/CollectionDetail/CollectionDetail.tsx [hunk 20]
```diff
@@ -1,51 +1,178 @@
...
+                  >
+                    {t('collection:go-to-ayah')}
+                  </PopoverMenu.Item>
+                </PopoverMenu>
+              }
             >
               {({ isOpen }) => {
                 if (!isOpen) return null;
-                return <div className={styles.item}>{item.content}</div>;
+                const chapterId = bookmark.key;
```
Comment (osamasayed) [inline src/components/Collection/CollectionDetail/CollectionDetail.tsx:139]: I think we should extract this to a new component as the file is becoming very bloated.

File: src/components/Collection/CollectionList/CollectionList.tsx [hunk 21]
```diff
@@ -1,76 +1,113 @@
...
+    { id: CollectionListSortOption.RecentlyUpdated, label: t('collection:recently-updated') },
+    { id: CollectionListSortOption.Alphabetical, label: t('collection:alphabetical') },
+  ];
+
+  if (!data) return null;
+
+  const collections = data?.data || [];
+
+  const onSortOptionChanged = (val) => {
+    setSortBy(val);
```
Comment (osamasayed) [inline src/components/Collection/CollectionList/CollectionList.tsx:48]: Add logging

File: src/components/Collection/CollectionList/RenameCollectionAction.tsx (hunk 8 above)
Comment (osamasayed) [inline src/components/Collection/CollectionList/RenameCollectionAction.tsx:14]: Add logging

File: src/components/Collection/CollectionList/DeleteCollectionAction.tsx (hunk 7 above)
Comment (osamasayed) [inline src/components/Collection/CollectionList/DeleteCollectionAction.tsx:9]: Add logging

File: src/components/Verse/OverflowVerseActionsMenuBody.tsx [hunk 22]
```diff
@@ -1,36 +1,21 @@
 /* eslint-disable react-func/max-lines-per-function */
 /* eslint-disable max-lines */
```
Comment (osamasayed) [inline src/components/Verse/OverflowVerseActionsMenuBody.tsx:2]: I think now that we cleaned up this file, those rules can be removed.

File: src/pages/collections/[collectionId]/index.tsx [hunk 23]
```diff
@@ -0,0 +1,122 @@
...
+import { isLoggedIn } from '@/utils/auth/login';
+import CollectionDetail from 'src/components/Collection/CollectionDetail/CollectionDetail';
+import Button from 'src/components/dls/Button/Button';
+import DataContext from 'src/contexts/DataContext';
+import { privateFetcher } from 'src/utils/auth/api';
+import { makeGetBookmarkByCollectionId } from 'src/utils/auth/apiPaths';
+import { getAllChaptersData } from 'src/utils/chapter';
+import { GetBookmarkCollectionsIdResponse } from 'types/auth/GetBookmarksByCollectionId';
+
+const defaultSortOptionId = 'recentlyAdded';
```
Comment (osamasayed) [inline src/pages/collections/[collectionId]/index.tsx:21]: This should be using the enum. Also I think we can remove it since it's used once and directly set as a default value inside `useState`.

File: src/pages/collections/[collectionId]/index.tsx [hunk 24]
```diff
@@ -0,0 +1,122 @@
...
+
+  useEffect(() => {
+    if (!isLoggedIn()) {
+      router.replace('/');
+    }
+  }, [router]);
+
+  const [sortBy, setSortBy] = useState(defaultSortOptionId);
+
+  const onSortByChange = (newSortByVal) => {
```
Comment (osamasayed) [inline src/pages/collections/[collectionId]/index.tsx:36]: Add logging

File: src/pages/collections/[collectionId]/index.tsx [hunk 25]
```diff
@@ -0,0 +1,122 @@
...
+    setSortBy(newSortByVal);
+  };
+
+  const getKey = (pageIndex, previousPageData) => {
```
Comment (osamasayed) [inline src/pages/collections/[collectionId]/index.tsx:40]: Would be awesome to add documentation to this method since it's a bit hard to follow.

File: src/pages/collections/[collectionId]/index.tsx [hunk 26]
```diff
@@ -0,0 +1,122 @@
...
+  };
+
+  const lastPageData = data[data.length - 1];
+  const { hasNextPage } = lastPageData.pagination;
+
+  const bookmarks = data.map((response) => response.data.bookmarks).flat();
+  const title = data[0].data.collection.name;
+
+  const loadMore = () => {
+    setSize(size + 1);
```
Comment (osamasayed) [inline src/pages/collections/[collectionId]/index.tsx:77]: Add logging

File: src/components/Verse/OverflowVerseActionsMenuBody.tsx [hunk 27]
```diff
@@ -231,21 +95,15 @@ const OverflowVerseActionsMenuBody: React.FC<Props> = ({
...
-      >
-        {isVerseBookmarked ? `${t('bookmarked')}!` : `${t('bookmark')}`}
-      </PopoverMenu.Item>
+      <BookmarkAction
+        verse={verse}
+        isTranslationView={isTranslationView}
+        onActionTriggered={onActionTriggered}
+        bookmarksRangeUrl={bookmarksRangeUrl}
+      />
+      <SaveToCollectionAction verse={verse} bookmarksRangeUrl={bookmarksRangeUrl} />
```
Comment (osamasayed) [inline src/components/Verse/OverflowVerseActionsMenuBody.tsx:104]: This needs to be conditionally rendered only when the user is logged in. Another idea is to encourage people to signin/up so when they click on it, we show a toast saying please sign in to use this feature.

File: src/components/Verse/SaveToCollectionAction.tsx [hunk 28]
```diff
@@ -0,0 +1,179 @@
...
+          });
+          return;
+        }
+        toast(t('common:error.general'), {
+          status: ToastStatus.Error,
+        });
+      });
+  };
+
+  const onNewCollectionCreated = (newCollectionName: string) => {
```
Comment (osamasayed) [inline src/components/Verse/SaveToCollectionAction.tsx:121]: Add logging

File: src/components/Verse/SaveToCollectionAction.tsx [hunk 29]
```diff
@@ -0,0 +1,179 @@
...
+import { isLoggedIn } from 'src/utils/auth/login';
+import { logButtonClick } from 'src/utils/eventLogger';
+import BookmarkType from 'types/BookmarkType';
+
+const SaveToCollectionAction = ({ verse, bookmarksRangeUrl }) => {
+  const [isSaveCollectionModalOpen, setIsSaveCollectionModalOpen] = useState(false);
+  const quranReaderStyles = useSelector(selectQuranReaderStyles, shallowEqual);
+  const mushafId = getMushafId(quranReaderStyles.quranFont, quranReaderStyles.mushafLines).mushaf;
+  const { t } = useTranslation();
+  const { data: collectionListData, mutate: mutateCollectionListData } = useSWR(
```
Comment (osamasayed) [inline src/components/Verse/SaveToCollectionAction.tsx:39]: I noticed we are using `useSWR` here but for `bookmarkCollectionIdsData` we are using `useSWRImmutable`. Is there a reason for that? Also for `bookmarkCollectionIdsData` we check first if the user is logged in vs here we don't

File: src/components/Collection/SaveToCollectionModal/SaveToCollectionModal.tsx [hunk 30]
```diff
@@ -8,45 +8,68 @@ import FormBuilder from '@/components/FormBuilder/FormBuilder';
...
   onCollectionToggled,
   onNewCollectionCreated,
+  onClose,
 }: SaveToCollectionModalProps) => {
   const [isAddingNewCollection, setIsAddingNewCollection] = useState(false);
   const { t } = useTranslation();
+  const toast = useToast();
+
+  const handleSubmit = (data) => {
+    onNewCollectionCreated(data.name)
```
Comment (osamasayed) [inline src/components/Collection/SaveToCollectionModal/SaveToCollectionModal.tsx:44]: Add logging

**Review by osamasayed — APPROVED** [review]
